### PR TITLE
(1127) Add case list filters

### PR DESCRIPTION
**Pull request by jsrobertson**
## Context

https://trello.com/c/OHWmFT0d/1127-add-filters-to-assess-case-list-m

## Changes in this PR
Added two filters to the top of the case list page.  This will likely be further improved (e.g. new partial for filters components) as we add more of the case list pages but struggling to think that far ahead at this stage 😄 

Have made the filters at least look presentable on smaller screens too, despite the table needing improvement.

No results error message to be done under a separate card once we have content and design.

## Screenshots of UI changes

localhost_3000_assess_referrals_case-list (1)

image

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: assets/scss/local.scss [hunk 1]
```diff
@@ -5,3 +5,39 @@
 .maintain-white-space {
   white-space: pre-wrap;
 }
+
+.filters {
```
Comment (jsrobertson) [inline assets/scss/local.scss:9]: [Déjà vu](https://github.com/ministryofjustice/digital-prison-services/blame/dc7cb06bc1d09cf7715eeeb331bb0216563e89b2/sass/_horizontal-content.scss#L33)

File: integration_tests/e2e/assess/caseList.cy.ts [hunk 2]
```diff
@@ -32,6 +30,39 @@ context('Referral case lists', () => {
...
+      const path = assessPaths.caseList.show({})
+      cy.visit(path)
+
+      const caseListPage = Page.verifyOnPage(CaseListPage, {
+        referralSummaries,
+      })
+      caseListPage.shouldContainTableOfReferralSummaries()
+
+      const programmeStrandSelectedValue = 'General offence'
+      const referralStatuSelectedValue = 'ASSESSMENT_STARTED'
```
Comment (Gweaton) [inline integration_tests/e2e/assess/caseList.cy.ts:50]: missing an `s` on `referralStatu` here (sorry)

**Review by Gweaton — APPROVED** [review]
Looks good to me - a couple of small comments.

I was also wondering if it's possible that when we don't select a status, we don't send that param through in the URL - is that to do with us checking for an undefined value in the query params rather than an empty string?

image

File: integration_tests/pages/page.ts [hunk 3]
```diff
@@ -51,6 +51,10 @@ export default abstract class Page {
     cy.get(`.govuk-radios__input[name="${name}"][value="${value}"]`).check()
   }
 
+  selectSelectItem(testId: string, value: string): void {
```
[the remark this answers is not in the document]
Comment (yndajas) [reply integration_tests/pages/page.ts:54]: `selectSelectedSelectItemSelection`

File: server/services/referralService.ts [hunk 4]
```diff
@@ -39,12 +39,13 @@ export default class ReferralService {
   async getReferralSummaries(
     username: Express.User['username'],
     organisationId: Organisation['id'],
+    filterValues?: { audience?: string; status?: string },
```
[the remark this answers is not in the document]
Comment (jsrobertson) [reply server/services/referralService.ts:42]: `CourseAudience['value']` is OK but we need to add an enum or new type for the capitalised referral status needed for the api.

**Comment by jsrobertson** [comment]
> Looks good to me - a couple of small comments.
> 
> I was also wondering if it's possible that when we don't select a status, we don't send that param through in the URL - is that to do with us checking for an undefined value in the query params rather than an empty string?
> 
> image

I know what you mean but it's the standard behaviour of a form with a get method.  I think the alternative would be to have a new post route which strips out the `undefined` params and redirects back with only the non `undefined` values in the URL.

**Pull request merged**